presto-main/src/main/java/com/facebook/presto/metadata/SessionPropertyManager.java

```diff
@@ -0,0 +1,382 @@
+        if (sessionProperty == null) {
+            throw new PrestoException(INVALID_SESSION_PROPERTY, "Unknown session property " + name);
+        }
+        return sessionProperty.getMetadata();
+    }
+
+    public List<SessionPropertyValue> getAllSessionProperties(Session session)
+    {
+        requireNonNull(session, "session is null");
+
+        List<SessionPropertyValue> properties = new ArrayList<>();
+        for (SessionProperty<?> sessionProperty : allSessionProperties.values()) {
+            SessionPropertyMetadata<?> propertyMetadata = sessionProperty.getMetadata();
+            String defaultValue = propertyMetadata.getDefaultValue() == null ? "" : propertyMetadata.getDefaultValue().toString();
+
+            String value;
+            if (sessionProperty.getCatalogName().isPresent()) {
+                value = session.getCatalogProperties(sessionProperty.getCatalogName().get()).get(sessionProperty.getPropertyName());
+            }
+            else {
+                value = session.getSystemProperties().get(sessionProperty.getPropertyName());
+            }
+
+            if (value == null) {
+                value = defaultValue;
+            }
+
+            properties.add(new SessionPropertyValue(
+                    value,
+                    defaultValue,
+                    sessionProperty.getFullyQualifiedName(),
+                    sessionProperty.getCatalogName(),
+                    sessionProperty.getPropertyName(),
+                    sessionProperty.getMetadata().getDescription(),
+                    sessionProperty.getMetadata().getSqlType().getTypeSignature().toString()));
+        }
+
+        // sort properties by catalog then property
+        Collections.sort(properties, (left, right) -> ComparisonChain.start()
+                .compare(left.getCatalogName().orElse(null), right.getCatalogName().orElse(null), Ordering.natural().nullsFirst())
+                .compare(left.getPropertyName(), right.getPropertyName())
+                .result());
+
+        return ImmutableList.copyOf(properties);
+    }
+
+    public <T> T decodeProperty(String name, @Nullable String value, Class<T> type)
+    {
+        requireNonNull(name, "name is null");
+        requireNonNull(type, "type is null");
+
+        SessionProperty<?> sessionProperty = allSessionProperties.get(name);
+        if (sessionProperty == null) {
+            throw new PrestoException(INVALID_SESSION_PROPERTY, "Unknown session property " + name);
+        }
+        SessionPropertyMetadata<?> metadata = sessionProperty.getMetadata();
+        if (metadata.getJavaType() != type) {
+            throw new PrestoException(INVALID_SESSION_PROPERTY, String.format("Property %s is type %s, but requested type was %s", name,
+                    metadata.getJavaType().getName(),
+                    type.getName()));
+        }
+        if (value == null) {
+            return type.cast(metadata.getDefaultValue());
```

Comment:
Definitely not. The default value is defined in terms of the final Java type which may not be String.  Currently, null is a legal default type, but not a legal user set value.